pkg/nfd-client/topology-updater/nfd-topology-updater.go

```diff
@@ -0,0 +1,241 @@
+		return err
+	}
+
+	klog.Infof("resAggr is: %v\n", resAggr)
+
+	// Create watcher for TLS certificates
+	w.certWatch, err = utils.CreateFsWatcher(time.Second, w.nodeInfo.args.CaFile, w.nodeInfo.args.CertFile, w.nodeInfo.args.KeyFile)
+	if err != nil {
+		return err
+	}
+
+	crTrigger := time.After(0)
+	for {
+		select {
+		case <-crTrigger:
+			klog.Infof("Scanning\n")
+			podResources, err := resScan.Scan()
+			utils.KlogDump(1, "podResources are", "  ", podResources)
+			if err != nil {
+				klog.Warningf("Scan failed: %v\n", err)
+				continue
+			}
+			zones = resAggr.Aggregate(podResources)
+			utils.KlogDump(1, "After aggregating resources identified zones are", "  ", zones)
+			if err = w.Update(zones); err != nil {
+				klog.Exit(err)
+			}
+
+			if w.nodeInfo.args.Oneshot {
+				return nil
+			}
+
+			if w.nodeInfo.resourcemonitorArgs.SleepInterval > 0 {
+				crTrigger = time.After(w.nodeInfo.resourcemonitorArgs.SleepInterval)
+			}
+
+		case <-w.certWatch.Events:
+			klog.Infof("TLS certificate update, renewing connection to nfd-master")
+			w.Disconnect()
+			if err := w.Connect(); err != nil {
+				return err
+			}
+
+		case <-w.stop:
+			klog.Infof("shutting down nfd-worker")
+			w.certWatch.Close()
+			return nil
+		}
+	}
+
+}
+
+func (w *nfdTopologyUpdater) Update(zones v1alpha1.ZoneList) error {
+	// Connect to NFD master
+	err := w.Connect()
+	if err != nil {
+		return fmt.Errorf("failed to connect: %v", err)
+	}
+	defer w.Disconnect()
+	// Update the node with the feature labels.
+	if w.client == nil {
+		return nil
+	}
+
+	err = advertiseNodeTopology(w.client, zones, w.nodeInfo.tmPolicy, nfdclient.NodeName())
+	if err != nil {
+		return fmt.Errorf("failed to advertise node topology: %s", err.Error())
+	}
+	return nil
+}
+
+// Stop NfdWorker
+func (w *nfdTopologyUpdater) Stop() {
+	select {
+	case w.stop <- struct{}{}:
+	default:
+	}
+}
+
+// connect creates a client connection to the NFD master
+func (w *nfdTopologyUpdater) Connect() error {
+	// Return a dummy connection in case of dry-run
+	if w.nodeInfo.args.NoPublish {
+		return nil
+	}
+
+	if err := w.NfdBaseClient.Connect(); err != nil {
+		return err
+	}
+	w.client = pb.NewNodeTopologyClient(w.ClientConn())
+
+	return nil
+}
+
+// disconnect closes the connection to NFD master
+func (w *nfdTopologyUpdater) Disconnect() {
+	w.NfdBaseClient.Disconnect()
+}
+
+// advertiseNodeTopology advertises the topology CR to a Kubernetes node
+// via the NFD server.
+func advertiseNodeTopology(client pb.NodeTopologyClient, zoneInfo v1alpha1.ZoneList, tmPolicy string, nodeName string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	zones := make([]*v1alpha1.Zone, len(zoneInfo))
+	for i, zone := range zoneInfo {
+		zones[i] = &v1alpha1.Zone{
+			Name:      zone.Name,
+			Type:      zone.Type,
+			Parent:    zone.Parent,
+			Resources: zone.Resources,
+			Costs:     zone.Costs,
+		}
+	}
```

Comment:
Since this is not a major blocker and  appears to be falling under the readability improvements, I think we could leave it to be handled as part of the code refactoring to be handled in a separate PR.  Let me know if that sounds reasonable (/cc @fromani)

Reply:
As with the other comment I had, could you add a `TODO:` note so that this comment is not lost in the void of closed PRs? 😺